src/hardware/voodoo.cpp

```diff
@@ -3008,7 +3008,33 @@ iterated W    = 18.32 [48 bits]
 **************************************************************************/
 
 static voodoo_state* v = nullptr;
-static uint8_t vtype = VOODOO_1, vperf;
+static uint8_t vtype   = VOODOO_1;
+
+enum class PerformanceFlags : uint8_t {
+	None                = 0,
+	MultiThreading      = 1 << 0,
+	NoBilinearFiltering = 1 << 1,
+	All                 = (MultiThreading | NoBilinearFiltering),
+};
+
+static const char* describe_performance_flags(const PerformanceFlags flags)
+{
+	// Note: the descriptions are meant to be used as a status postfix
+	switch (flags) {
+	case PerformanceFlags::None:
+		return " and no optimizations";
+	case PerformanceFlags::MultiThreading:
+		return " and multi-threading";
+	case PerformanceFlags::NoBilinearFiltering:
+		return " and no bilinear filtering";
+	case PerformanceFlags::All:
+		return ", multi-threading, and no biliear filtering";
+	}
+	assert(false);
+	return "unknown performance flags (bug)";
```

Comment:
Maybe drop "(bug)" 😄 (we don't do that anywhere else in this "safety" error messages)

Reply:
yes :-)
Dropped!